torch/nn/quantized/functional.py

```diff
@@ -0,0 +1,14 @@
+import types
+from torch._ops import ops
+
+class _Functional(types.ModuleType):
+    def __init__(self):
+        super(_Functional, self).__init__('torch.nn.quantized.functional')
+
+    def __getattr__(self, op_name):
+        import sys
+        op = ops.quantized.__getattr__(op_name)
+        setattr(self, op_name, op)
+        return op
+
+functional = _Functional()
```

Comment:
can't we just do functional = torch.ops.quantized ?

also, it'd be nice to add `__dir__` method on ops.py, but it might be a separate todo (create an issue?)

Reply:
If that works we should do that.

Reply:
functional = torch.ops.quantized does not work for me, that's why I changed to this.

Reply:
@dzhulgakov for `__dir__` do you mean we want it to print all the exposed namespace/method? but I think these are created dynamically, will `__dir__` still be useful?

Reply:
this isn't currently a good user experience.

For example:
```
>>> import torch.nn.quantized.functional as F
>>> F
Out[2]: ---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/data/users/gchanan/anaconda3/envs/__py36f/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

/data/users/gchanan/anaconda3/envs/__py36f/lib/python3.6/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    400                         if cls is not object \
    401                                 and callable(cls.__dict__.get('__repr__')):
--> 402                             return _repr_pprint(obj, self, cycle)
    403
    404             return _default_pprint(obj, self, cycle)

/data/users/gchanan/anaconda3/envs/__py36f/lib/python3.6/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    695     """A pprint that just redirects to the normal repr function."""
    696     # Find newlines and replace them with p.break_()
--> 697     output = repr(obj)
    698     for idx,output_line in enumerate(output.splitlines()):
    699         if idx:

/data/users/gchanan/anaconda3/envs/__py36f/lib/python3.6/importlib/_bootstrap.py in _module_repr(module)

/data/users/gchanan/_pytorch6/torch/nn/quantized/functional.py in __getattr__(self, op_name)
      7
      8     def __getattr__(self, op_name):
----> 9         op = ops.quantized.__getattr__(op_name)
     10         setattr(self, op_name, op)
     11         return op

/data/users/gchanan/_pytorch6/torch/_ops.py in __getattr__(self, op_name)
     58         # for overloads and raise an exception if there are more than one.
     59         qualified_op_name = '{}::{}'.format(self.name, op_name)
---> 60         op = torch._C._jit_get_operation(qualified_op_name)
     61         # let the script frontend know that op is identical to the builtin op
     62         # with qualified_op_name

RuntimeError: No such operator quantized::__file__
```

```
>>> dir(F)
['__doc__', '__loader__', '__name__', '__package__', '__spec__']
```

Compared to NN:
```
>>> import torch.nn.functional as F

>>> F
Out[5]: <module 'torch.nn.functional' from '/data/users/gchanan/_pytorch6/torch/nn/functional.py'>

>>> dir(F)
Out[6]:
['GRID_SAMPLE_INTERPOLATION_MODES',
 'GRID_SAMPLE_PADDING_MODES',
 'List',
 '_Reduction',
 '_VF',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 '_adaptive_max_pool1d',
 '_adaptive_max_pool2d',
 '_adaptive_max_pool3d',
 '_add_docstr',
 '_fractional_max_pool2d',
 '_fractional_max_pool3d',
 '_get_softmax_dim',
 '_infer_size',
 '_list_with_default',
 '_max_pool1d',
 '_max_pool2d',
 '_max_pool3d',
 '_no_grad_embedding_renorm_',
 '_pad_circular',
 '_pair',
 '_pointwise_loss',
 '_single',
 '_smooth_l1_loss',
 '_triple',
 '_unpool_output_size',
 'adaptive_avg_pool1d',
 'adaptive_avg_pool2d',
 'adaptive_avg_pool3d',
 'adaptive_max_pool1d',
 'adaptive_max_pool1d_with_indices',
 'adaptive_max_pool2d',
 'adaptive_max_pool2d_with_indices',
 'adaptive_max_pool3d',
 'adaptive_max_pool3d_with_indices',
 'affine_grid',
 'alpha_dropout',
 'assert_int_or_pair',
 'avg_pool1d',
 'avg_pool2d',
 'avg_pool3d',
 'batch_norm',
 'bilinear',
 'binary_cross_entropy',
 'binary_cross_entropy_with_logits',
 'celu',
 'celu_',
 'conv1d',
 'conv2d',
 'conv3d',
 'conv_tbc',
 'conv_transpose1d',
 'conv_transpose2d',
 'conv_transpose3d',
 'cosine_embedding_loss',
 'cosine_similarity',
 'cross_entropy',
 'ctc_loss',
 'division',
 'dropout',
 'dropout2d',
 'dropout3d',
 'elu',
 'elu_',
 'embedding',
 'embedding_bag',
 'feature_alpha_dropout',
 'fold',
 'fractional_max_pool2d',
 'fractional_max_pool2d_with_indices',
 'fractional_max_pool3d',
 'fractional_max_pool3d_with_indices',
 'glu',
 'grad',
 'grid_sample',
 'group_norm',
 'gumbel_softmax',
 'hardshrink',
 'hardtanh',
 'hardtanh_',
 'hinge_embedding_loss',
 'instance_norm',
 'interpolate',
 'kl_div',
 'l1_loss',
 'layer_norm',
 'leaky_relu',
 'leaky_relu_',
 'linear',
 'local_response_norm',
 'log_softmax',
 'logsigmoid',
 'lp_pool1d',
 'lp_pool2d',
 'margin_ranking_loss',
 'math',
 'max_pool1d',
 'max_pool1d_with_indices',
 'max_pool2d',
 'max_pool2d_with_indices',
 'max_pool3d',
 'max_pool3d_with_indices',
 'max_unpool1d',
 'max_unpool2d',
 'max_unpool3d',
 'mse_loss',
 'multi_margin_loss',
 'multilabel_margin_loss',
 'multilabel_soft_margin_loss',
 'nll_loss',
 'normalize',
 'one_hot',
 'pad',
 'pairwise_distance',
 'pdist',
 'pixel_shuffle',
 'poisson_nll_loss',
 'prelu',
 'relu',
 'relu6',
 'relu_',
 'rrelu',
 'rrelu_',
 'selu',
 'selu_',
 'sigmoid',
 'smooth_l1_loss',
 'soft_margin_loss',
 'softmax',
 'softmin',
 'softplus',
 'softshrink',
 'softsign',
 'tanh',
 'tanhshrink',
 'threshold',
 'threshold_',
 'torch',
 'triplet_margin_loss',
 'unfold',
 'upsample',
 'upsample_bilinear',
 'upsample_nearest',
 'utils',
 'vision',
 'warnings',
 'weak_script']
```

Reply:
If `torch.ops.quantized` can't provide a `__dir__` I guess you'd need to manually bind the ops in here when the module is imported.

Reply:
OK looking at nn.functional.py I guess we'll just manually bind each op